### PR TITLE
Wal 59 port bookie common message component

### PR DESCRIPTION
**Pull request by aametzler**

### CODE REVIEW

File: src/store/configureStore.dev.js [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
+    actionsWhitelist = ['COMMON_MSG_ADD_MSG',
+      'COMMON_MSG_REMOVE_MSG'];
+  }
+
+  // Configure enhancer for redux dev tools extensions (if available)
+  const composeEnhancers = composeWithDevTools({
+    features: {
+      dispatch: true
+    },
+    // Option for immutable
```
Comment (mseaward) [inline src/store/configureStore.dev.js:29]: the commented code, are they needed here? If not, remove.

File: src/store/configureStore.dev.js [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+import Config from '../../config/Config';
 import thunk from 'redux-thunk';
 // import createLogger from 'redux-logger';
+import {composeWithDevTools} from 'redux-devtools-extension';
+import {autoRehydrate} from 'redux-persist-immutable';
 import rootReducer from '../reducers';
 // import DevTools from 'containers/DevTools';
 import {hashHistory} from 'react-router';
 import {routerMiddleware} from 'react-router-redux';
+//import * as actionCreators from '../actions/CommonMessageActions';
```
Comment (mseaward) [inline src/store/configureStore.dev.js]: if not used, remove.

File: src/reducers/CommonMessageReducer.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import Immutable from 'immutable';
+import ActionTypes from '../constants/ActionTypes';
+
+let initialState = Immutable.fromJS({
+  messageCount: 0,
+  headerMessages: [],
+  sideBarMessages: [],
```
Comment (mseaward) [inline src/reducers/CommonMessageReducer.js:7]: `sidebar` would make more sense being renamed `aside`.

File: src/reducers/CommonMessageReducer.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      return newState;
+    }
+
+    case ActionTypes.COMMON_MSG_REMOVE_MSG: {
+      // get current state of messages
+      // operate on the one matching supplied id
+      let messageCount = state.get('messageCount');
+      const newMessageCount = --messageCount;
+      const id = action.id;
+      const newExchangeMsgState = state.get('headerMessages')
```
Comment (mseaward) [inline src/reducers/CommonMessageReducer.js]: `newHeaderMsgState`

File: src/reducers/CommonMessageReducer.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        .filter((m) => {
+          let mID = m.get('id');
+
+          // Filter exchange messages.
+          return mID === 'h' + id;
+        });
+      const newsideBarMessagestate = state.get('sideBarMessages')
```
Comment (mseaward) [inline src/reducers/CommonMessageReducer.js:44]: const `newAsideMessagestate` = state.get('asideMessages')

These changes might require changes elsewhere in the implementation of this component.

File: src/reducers/CommonMessageReducer.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        .filter((m) => {
+          let mID = m.get('id');
+
+          // Filter betslip messages.
+          return mID === 's' + id;
```
Comment (mseaward) [inline src/reducers/CommonMessageReducer.js:49]: // Filter aside messages.
return mID === 'a' + id;

File: src/reducers/CommonMessageReducer.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        });
+
+      return state.set('headerMessages', newExchangeMsgState)
```
Comment (mseaward) [inline src/reducers/CommonMessageReducer.js]: return state.set('headerMessages', newHeaderMsgState)

**Review by mseaward — CHANGES_REQUESTED** [review]

**Comment by mseaward** [comment]
Move style sheet for the CommonMessage component into its own directory similarly to BookiePro.

File: src/components/CommonMessage/CommonMessage.jsx [hunk 8]
```diff
@@ -0,0 +1,156 @@
+import React, {PureComponent} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import CommonMessageActions from '../../actions/CommonMessageActions';//added
+import Config from '../../../config/Config';//added
+import MessageType from '../../constants/MessageTypes';//added
+import _ from 'lodash';
```
Comment (mseaward) [inline src/components/CommonMessage/CommonMessage.jsx:7]: what are these comments `//added` there for?

File: src/components/Dashboard/Balances/BalancesContainer.jsx [hunk 9]
```diff
@@ -1,13 +1,17 @@
 import React from 'react';
 import Side from './Side';
 import Balances from './Balances';
+import CommonMessage from '../../CommonMessage/CommonMessage';
```
Comment (mseaward) [inline src/components/Dashboard/Balances/BalancesContainer.jsx]: Not sure this level of specificity is required to import CommonMessage with the index.js within its directry.

**Review by mseaward — DISMISSED** [review]

**Review by mseaward — DISMISSED** [review]

**Review by imarquette — DISMISSED** [review]

File: src/components/Dashboard/Balances/Balances.jsx [hunk 10]
```diff
@@ -131,7 +131,7 @@ const mapStateToProps = (state) => {
     precision: dashboard.precision,
     decimals: dashboard.decimals,
     showHiddenAssets: dashboard.showHiddenAssets,
-    hiddenAssets: state.settings.hiddenAssets
+    hiddenAssets: state.settings.hiddenAssets,
```
Comment (mseaward) [inline src/components/Dashboard/Balances/Balances.jsx]: that comma needs to be killed ;p

File: src/components/Dashboard/Balances/Balances.jsx [hunk 11]
```diff
@@ -142,7 +142,7 @@ const mapDispatchToProps = (dispatch) => bindActionCreators(
     toggleAssetHidden: DashboardPageActions.toggleAssetHidden,
     toggleShowHiddenAssets: DashboardPageActions.toggleShowHiddenAssets,
     navigateToSend: NavigateActions.navigateToSend,
-    navigateToDepositWithDraw: NavigateActions.navigateToDepositWithDraw
+    navigateToDepositWithDraw: NavigateActions.navigateToDepositWithDraw,
```
Comment (mseaward) [inline src/components/Dashboard/Balances/Balances.jsx]: same for this comma

File: src/components/Header/Header.jsx [hunk 12]
```diff
@@ -100,7 +100,7 @@ const mapStateToProps = (state) => {
     currentAccount : state.account.currentAccount,
     starredAccounts : state.account.starredAccounts,
     locked : state.wallet.locked,
-    current_wallet : state.wallet.currentWallet
+    current_wallet : state.wallet.currentWallet,
```
Comment (mseaward) [inline src/components/Header/Header.jsx]: and another 🗡

**Review by mseaward — CHANGES_REQUESTED** [review]
see my comments, some commas to kill. Otherwise, looks good.

**Review by mseaward — APPROVED** [review]

**Pull request merged**